### PR TITLE
api: add name into filter chain

### LINKED ISSUE #7870: named filter chains
It would be good from a stats perspective to have names for the filter chains so that users can easily see how many connections matched on a specific filter chain match [say by SNI or ips]. From a FCDS perspective, adding names is also helpful as we can now tender filterchain resources as named objects, spit out informative errors for specific filter chains, etc. 

cc @silentdai ..

### PR DESCRIPTION
**Pull request by lambdai**
Signed-off-by: Yuchen Dai <silentdai@gmail.com>

fix #7870 #7867 

Description:
Add a field `name` into message `FilterChain`.
[ ] facilitate the name in istio to identify a filter chain.
[ ] add stat to record the hit number
[ ] eventually used by FCDS

Risk Level:
Testing:
Docs Changes:
Release Notes:

### CODE REVIEW

File: api/envoy/api/v2/listener/listener.proto [hunk 1]
```diff
@@ -188,6 +188,11 @@ message FilterChain {
 
   // See :ref:`base.TransportSocket<envoy_api_msg_core.TransportSocket>` description.
   core.TransportSocket transport_socket = 6;
+
+  // The unique name (or empty) by which this filter chain is known. If no name is provided,
+  // Envoy will allocate an internal UUID for the listener. If the listener is to be dynamically
```
Comment (mattklein123) [inline api/envoy/api/v2/listener/listener.proto]:
```suggestion
  // Envoy will allocate an internal UUID for the filter chain. If the filter chain is to be dynamically
```

File: api/envoy/api/v2/listener/listener.proto [hunk 2]
```diff
@@ -188,6 +188,11 @@ message FilterChain {
...
+  // updated or removed via FCDS(not implemented yet) a unique name must be provided.
```
Comment (mattklein123) [inline api/envoy/api/v2/listener/listener.proto]:
```suggestion
  // updated or removed via FCDS a unique name must be provided.
```

Please mark this field not-implemented-hide until it's actually implemented.

File: api/envoy/api/v2/listener/listener.proto (hunk 1 above)
Comment (lambdai) [reply api/envoy/api/v2/listener/listener.proto]: oops

File: api/envoy/api/v2/listener/listener.proto (hunk 2 above)
Comment (htuch) [reply api/envoy/api/v2/listener/listener.proto]: +1 on this

**Comment by lambdai** [comment]
fix and force pushed

**Pull request merged**